bokeh/models/renderers.py

```diff
@@ -168,6 +169,46 @@ def _check_bad_column_name(self):
 
     level = Override(default="glyph")
 
+_DEFAULT_NODE = lambda: Circle()
+
+_DEFAULT_EDGE = lambda: MultiLine()
+
+class GraphRenderer(DataRenderer):
+    '''
+
+    '''
+
+    x_range_name = String('default', help="""
+    A particular (named) x-range to use for computing screen
+    locations when rendering glyphs on the plot. If unset, use the
+    default x-range.
+    """)
+
+    y_range_name = String('default', help="""
+    A particular (named) y-range to use for computing screen
+    locations when rendering glyphs on the plot. If unset, use the
+    default y-range.
+    """)
+
+    graph_source = Instance(GraphSource, help="""
+    Instance of a GraphSource which contains the information about the
+    nodes and edges of a network graph.
+    """)
+
+    nodes = Instance(Marker, default=_DEFAULT_NODE, help="""
+    Instance of a Marker glyph that will be rendered as the graph nodes. The
+    marker attribute fields will be matched against columns in the
+    ``graph_source.nodes`` ColumnDataSource.
+    """)
+
+    edges = Instance(MultiLine, default=_DEFAULT_EDGE, help="""
+    Instance of a MultiLine glyph that will be rendered as the graph edges. The
+    multiline attribute fields will be matched against columns in the
+    ``graph_source.edges`` ColumnDataSource.
```

Comment:
commenting for posterity: The MultiLine glyph is exposed so that users can style the visual attributes of the graph edges. The LayoutProvider will supply the coordinate data.